### PR TITLE
Bg order status

### LINKED ISSUE #127: Order status Not changing to 'Packaged' US 31
When the last item in the order is fulfilled. there is a strange empty flash message on the page and the order is not marked at 'packaged'. the order items are correctly marked as fulfilled but the entire order status does not change

### PR DESCRIPTION
**Pull request by brian-greeson**
## Order status now updates correctly
Closes #127 
- Expanded item order fulfill test to include failing state
- item order fulfill method now returns true if successful
- Added helper methods to Order, set_order_packaged and canceled?

### CODE REVIEW

File: app/controllers/merchant/item_orders_controller.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@ def update
     if item_order.fulfill
       flash[:success] = "#{item_order.item.name} has been fulfilled"
     else
-        flash[:error] = item_order.errors.full_messages.to_sentence
+      flash[:error] = item_order.errors.full_messages.to_sentence
```
Comment (fredrondina) [inline app/controllers/merchant/item_orders_controller.rb:8]: nice formatting change here

File: app/models/order.rb [hunk 2]
```diff
@@ -21,6 +21,14 @@ def cancel
     self.update(status: "canceled")
   end
 
+  def set_order_packaged
+    return false if canceled?
+    update(status: "packaged") unless item_orders.where(status: "unfulfilled").any?
```
Comment (fredrondina) [inline app/models/order.rb:26]: nice guard here

**Review by fredrondina — APPROVED** [review]
looks good to me, nice work

File: app/models/item_order.rb [hunk 3]
```diff
@@ -32,7 +33,9 @@ def unfulfilled?
   end
 
   def items
-    ItemOrder.where('order_id = ?', order.id)
+    binding.pry
+    order.order_items
+    # ItemOrder.where('order_id = ?', order.id)
```
Comment (kelshadarby) [inline app/models/item_order.rb]: You still have a pry and commented out line here

**Review by kelshadarby — CHANGES_REQUESTED** [review]
app/models/item_order.rb line 34 has some stuff I think you forgot to take out

File: app/models/item_order.rb (hunk 3 above)
Comment (fredrondina) [reply app/models/item_order.rb]: good catch

File: app/models/item_order.rb [hunk 4]
```diff
@@ -32,7 +33,9 @@ def unfulfilled?
   end
 
   def items
-    ItemOrder.where('order_id = ?', order.id)
+    binding.pry
```
Comment (fredrondina) [inline app/models/item_order.rb]: few things here to remove

**Review by fredrondina — CHANGES_REQUESTED** [review]

File: app/models/item_order.rb (hunk 3 above)
Comment (brian-greeson) [reply app/models/item_order.rb]: excellent, i'll get that removed

**Review by fredrondina — APPROVED** [review]

File: app/models/item_order.rb (hunk 4 above)
Comment (brian-greeson) [reply app/models/item_order.rb]: done

**Review by kelshadarby — APPROVED** [review]

**Pull request merged**